crates/ruff_linter/src/rules/pylint/rules/consider_swap_variables.rs

```diff
@@ -0,0 +1,151 @@
+use ruff_diagnostics::{Applicability, Edit, Fix};
+use ruff_python_ast::Stmt;
+use ruff_python_ast::name::Name;
+use ruff_text_size::{Ranged, TextRange, TextSize};
+
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+
+use crate::checkers::ast::Checker;
+use crate::{FixAvailability, Violation};
+
+/// ## What it does
+/// Checks for code that swaps two variables using a temporary variable.
+///
+/// ## Why is this bad?
+/// Variables can be swapped by using "tuple unpacking" instead of using a temporary variable. That also makes the intention of the swapping logic more clear.
+///
+/// ## Example
+/// ```python
+/// def function(x, y):
+///     if x > y:
+///         temp = x
+///         x = y
+///         y = temp
+///     assert x <= y
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def function(x, y):
+///     if x > y:
+///         x, y = y, x
+///     assert x <= y
+/// ```
+///
+/// ## Fix safety
+/// The rule's fix is marked as safe, unless it contains comments. In this
+/// exception case, applying the quick fix would remove comments between the
+/// assignment statements.
+#[derive(ViolationMetadata)]
+#[violation_metadata(preview_since = "0.0.0")]
```

Comment:
Let's go ahead and put the next patch version here:


```suggestion
#[violation_metadata(preview_since = "0.14.11")]
```